File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        for(int i = 0; i < s.length() ; i++){
            char current = s.charAt(i);
            
            if(current == '('){
                stack.push(current);
            } else {
                if(stack.isEmpty()){
                    answer = false;
                } else {
                    if(stack.peek() == '(') {
                        stack.pop();
                    } else {
                        answer = false;
                    }
                }
            }
        }
        if(!stack.isEmpty()) answer = false;
        return answer;
    }
}

File: JavaScript/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    // let count = 0;
    // for(const e of s){
    //     if(e === '('){
    //         count++;
    //     } else {
    //         if(count === 0) return false;
    //         count--;
    //     }
    // }
    // return count === 0;
    return correctBracket(s);
}

function correctBracket(string){
        const stack = [];
        for(let i = 0 ; i < string.length; i++){
            const bracket = string[i];
            if(bracket === "("){
                stack.push(bracket);
            } else { // ")"
                if(stack.length === 0){
                    return false;
                } else {
                    stack.pop();
                }
            }
        }
        return stack.length === 0;
    }

